### DISCUSSION TITLE
any info / basic example on how to do a multi Tennant ( saas ) application with remix?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [remix-run/remix#4311] [Q&A]
Any info / basic example on how to do a multi Tennant application with remix?

I have been looking for a basic simple example but unable to find one for remix.
Anyone has written about this? I have searched google, medium.com blog posts; nothing usable comes up.

My idea is to start toying with remix and build the foundation for a multi Tennant login setup ( saas alike ) 

```I would be able to login to a 'dashboard' that shows some placeholders and would display data from each Tennant only.```

I found several commercial saas products but they are way to bloated and I want to start from scratch implement just the features I need like auth and some basics nothing fancy to learn the basics of remix also. Using probably remix / prisma. But if more advanced opensource database platforms like supabase offer a multi Tennant solution im interested in that regard.

Any thought regarding this?

**Comment by dusty** [comment]
I'm doing this in a way currently using express.  I have a middleware do a lookup based on the hostname, then that is passed to context.  If the hostname isn't found a 404 is thrown.

Then inside remix all the loaders and actions can pull this data from context.

Any specific questions about that that might interest you?

**Reply by dusty** [reply]
Or are you just asking about the database setup?  As in a specific db for each tenant vs a column identifier.

**Reply by Jarrodsz** [reply]
Im looking for an example that implements tennant_in single database setup. 
Or another solution. 

Context is something that you use in your database model I think? It's like a workspace where team members would be added.

Your auth uses the context to get the correct tenants details based on the 'context' you mentioned?

Maybe it is really simple in remix / express /node but I don't know why there are blog posts regarding this. 
Multi Tennant / saas seems very hot issue.
